app/channel.go

```diff
@@ -3030,3 +3051,116 @@ func (a *App) getDirectChannel(userID, otherUserID string) (*model.Channel, *mod
 
 	return channel, nil
 }
+
+func (a *App) isUserMentionedInPost(team *model.Team, channel *model.Channel, post *model.Post, user *model.User, channelMemberNotifyPropsMap map[string]model.StringMap) (bool, error) {
```

Comment:
This is method that looks like it's a simple check, but in fact has a whole bunch of side effects (it updates things in passed arguments). Is this an accepted pattern in our codebase? It sounds like potentially confusing (generally I'd hope an `isX` style method would have no side effects).

Reply:
@zefhemel - not sure what you mean. This method has no side effects and doesn't update any of the parameters. (unless I missed something)

Reply:
Somehow I thought I saw you were changing state, but now looking at it again you're not. Ignore me.